packages/edit-site/src/components/sidebar/global-styles-v2/screens/typography-element.js

```diff
@@ -0,0 +1,286 @@
+	Slider,
+	TextInput,
+	UnitInput,
+	useNavigatorParams,
+	VStack,
+} from '@wordpress/components';
+import {
+	formatCapitalize,
+	formatLowercase,
+	formatUppercase,
+} from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { Screen, ScreenHeader } from '../components';
+import { useAppState } from '../state';
+
+const typographyOptions = [
+	{
+		label: 'Font',
+		value: 'fontFamily',
+		defaultValue: 'Helvetica Neue',
+	},
+	{
+		label: 'Size',
+		value: 'fontSize',
+		defaultValue: '14px',
+	},
+	{
+		label: 'Appearance',
+		value: 'fontWeight',
+		defaultValue: '400',
+	},
+	{
+		label: 'Line height',
+		value: 'lineHeight',
+		defaultValue: '1.5',
+	},
+	{
+		label: 'Letter spacing',
+		value: 'letterSpacing',
+		defaultValue: '0',
+	},
+	{
+		label: 'Letter case',
+		value: 'letterCase',
+		defaultValue: null,
+	},
+];
+
+const TypographyOptions = () => {
+	const {
+		attributes: { styles },
+		toggleAttribute,
+	} = useStyleAttributesForScreen();
+
+	const options = typographyOptions.map( ( o ) => {
+		return {
+			...o,
+			isSelected: ! isUndefined( styles[ o.value ] ),
+			onClick: () => toggleAttribute( o.value, o.defaultValue ),
+		};
+	} );
+
+	return (
+		<Dropdown>
+			<DropdownTrigger icon="ellipsis" isControl isSubtle size="small" />
+			<DropdownMenu>
+				{ options.map( ( option ) => (
+					<DropdownMenuItem key={ option.value } { ...option }>
+						{ option.label }
+					</DropdownMenuItem>
+				) ) }
+			</DropdownMenu>
+		</Dropdown>
+	);
+};
+
+const Header = ( { title } ) => {
+	return (
+		<ScreenHeader
+			action={ <TypographyOptions /> }
+			back="/typography"
+			description={ `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla
+				congue finibus ante vel maximus.` }
+			title={ title }
+		/>
+	);
+};
+
+function useStyleAttributesForScreen() {
+	const { id } = useNavigatorParams();
+	const appState = useAppState();
+	const { get, set } = appState;
+
+	let index;
+	const attributes = appState.typography.elements.find( ( e, i ) => {
+		if ( e.slug === id ) {
+			index = i;
+			return true;
+		}
+		return false;
+	} );
+
+	const getPath = ( attr ) =>
+		`typography.elements[${ index }].styles.${ attr }`;
+
+	const getAttribute = ( attr ) => attributes.styles[ attr ];
+	const setAttribute = ( attr ) => ( next ) => {
+		const path = getPath( attr );
+		const value = isUndefined( next ) ? null : next;
+		set( path, value );
+	};
+	const toggleAttribute = ( attr, defaultValue = undefined ) => {
+		const path = getPath( attr );
+		const current = get( path );
+		if ( isUndefined( current ) ) {
+			set( path, defaultValue );
+		} else {
+			set( path, undefined );
+		}
+	};
+
+	const getIsDefined = ( attr ) => {
+		const path = getPath( attr );
+		const current = get( path );
+		return ! isUndefined( current );
+	};
+
+	return {
+		getAttribute,
+		setAttribute,
+		toggleAttribute,
+		getIsDefined,
+		attributes,
+	};
+}
+
+const fontWeights = [ 100, 200, 300, 400, 500, 600, 700, 800, 900 ];
```

Comment:
We'll need a more central location for those.

Reply:
Currently, only appearance control component is aware of the weights gutenberg/packages/block-editor/src/components/font-appearance-control/index.js. I guess we can keep the same logic and just update that component for the new design (it is experimental).